third_party/skylib/repos.bzl

```diff
@@ -0,0 +1,22 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")
+
+def bazel_skylib_repos(local = None):
+    if local:
+        native.local_repository(
+            name = "bazel_skylib",
+            path = local,
+        )
+    else:
+        maybe(
+            http_archive,
+            name = "bazel_skylib",
+            sha256 = "cd55a062e763b9349921f0f5db8c3933288dc8ba4f76dd9416aac68acee3cb94",
+            urls = [
+                "https://github.com/bazelbuild/bazel-skylib/releases/download/1.5.0/bazel-skylib-1.5.0.tar.gz",
+            ],
+        )
```

Comment:
Will `http_archive_or_local` work here?

```
load("//rules:repo.bzl", "http_archive_or_local")

def bazel_skylib_repos(local = None):
  maybe(
    http_archive_or_local,
    name = "bazel_skylib",
    ...
    local = local,
  )
```

Reply:
unfortunately, wrapping the maybe around `http_archive_or_local` produces:
```
Cycle in the workspace file detected. This indicates that a repository is used prior to being defined.
The following chain of repository dependencies lead to the missing definition.
 - @bazel_skylib
This could either mean you have to add the '@bazel_skylib' repository with a statement like `http_archive` in your WORKSPACE file (note that transitive dependencies are not added automatically), or move an existing definition earlier in your WORKSPACE file.
```